File: gestao-angular/src/app/shared/dto/filterDTO/funcionario.filter.dto.ts
import {PessoaFilterDTO} from "./pessoa.filter.dto";

export class FuncionarioFilterDTO{

  constructor() {
    this.pessoaFilterDTO = new PessoaFilterDTO();
  }

  pessoaFilterDTO: PessoaFilterDTO;

  admissao: Date;

  admissaoInicio: Date;

  admissaoFim: Date;

}

File: gestao-spring/src/main/java/com/erp/gestao/dto/filterDTO/FuncionarioFilterDTO.java
package com.erp.gestao.dto.filterDTO;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
public class FuncionarioFilterDTO  {

    private PessoaFilterDTO pessoaFilterDTO;

    private LocalDate admissao;

    private LocalDate admissaoInicio;

    private LocalDate admissaoFim;

}
